flake.nix

```diff
@@ -52,7 +52,24 @@
             emanote = addBuildDepends super.emanote [ config.packages.stork ];
           };
         };
-        packages.default = config.packages.emanote;
+        packages.default =
+          let
+            haskellExeSansDependencyBloat = pkg: get-deps: with pkgs.haskell.lib;
+              (justStaticExecutables pkg).overrideAttrs (old: rec {
+                disallowedReferences = get-deps config.haskellProjects.default.haskellPackages;
+                # Ditch data dependencies that are not needed at runtime.
+                # cf. https://github.com/NixOS/nixpkgs/pull/204675
+                postInstall = (old.postInstall or "") + ''
+                  ${lib.concatStrings (map (e: "remove-references-to -t ${e} $out/bin/emanote\n") disallowedReferences)}
```

Comment:
Is there a way to *not hardcode* `$out/bin/emanote` here?

Reply:
Because I'm wondering if this can be upstreamed to https://github.com/srid/haskell-flake (cc @roberth) in some form to make it useful for other projects.

Reply:
`$out/bin/*` perhaps? However I don't think we should need it. Why are unused references created in the first place?

Reply:
If this is about `data-files`, maybe you need `extra-source-files` instead?

Reply:
With the new changes, I suppose another idea is to just upstream `removeReferencesTo` to nixpkgs. That or wait for that cabal PR to go through.

Reply:
You'd still have to manually specify all the deps, and look at this example to understand context. I figure at that point you can just copy the one-liner over and edit.